### ISSUE TITLE
Configure plugins clients with plugins

### ISSUE THREAD

**Issue by Nyholm** [php-http/HttplugBundle#15]
Can I get some feedback on this approach to configure the plugins?

```yaml
my_redirect_plugin:
  class: Http\Client\Plugin\RedirectPlugin 
my_logger_plugin:
  class: Http\Client\Plugin\LoggerPlugin 
  arguments: ["@monolog"]
my_cache_plugin:
  class: Http\Client\Plugin\CachePlugin 
  arguments: ["@cache", "@stream_factory"]

httplug:
  clients: 
    my_guzzle5: 
      factory: 'httplug.factory.guzzle5'
      plugins: ['@my_logger_plugin', '@my_cache_plugin']
      config:
        defaults:
          verify_ssl: false
          timeout: 4
    acme: 
      factory: 'httplug.factory.guzzle6'
      plugins: ['@my_logger_plugin', '@my_redirect_plugin']
      config:
        base_uri: 'http://google.se/'
```

This would allow you to use your own plugins, set the order etc. It would also be easy to inject a JournalPlugin to use internally for the debug toolbar. 

We could go with an approach where we name the plugins etc etc but it is a simplier implementation and easier to use (no magic) if we reference them as a service.

**Comment by dbu** [comment]
this looks simple and clean enough, i like it!

maybe we can provide default services for the plugins provided by php-http/plugin, if needed with options to configure things on them. but as with client factories, you would just use the full service name php_http.plugin.redirect (or similar) to add the plugin to a client, no shortcut that adds magic.

**Comment by Nyholm** [comment]
Closed by #16
